### PR TITLE
New Practice Git Flow

### PR DESCRIPTION
**Pull request by Kristynaslomkova**

### CODE REVIEW

File: practices/git_flow.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+4. Merge the feature branch back into develop branch when the feature is finished.
+5. Create a release branch when the features are ready for a release. The parent is the develop branch.
+6. Merge the release branch into the master branch and tag it with a version number. After that, it is merged back into the develop branch so the progress since the release is saved. The release branch will be deleted.
+7. If you need to make some critical changes, create a hotfix branch. Fork the branch directly off of the master branch. When you are finished with the changes, merge the hotfix branch to both master and develop branch (or the current release branch). Tag the master branch with an updated version number.
+
+## Common Pitfalls of the Git Flow Workflow
+- Not everyone in the team uses Git Flow Workflow. It is crucial for smooth project development to ensure that all the team understands and uses the same workflow.
+- The Git Flow Workflow is customized for a specific product. The setting is too complicated for a new developer.
+- The team has too many features and is unable to finish them before the release. Development is drifting away from production.
```
Comment (pamichalik) [inline practices/git_flow.md]: Tady by bylo dobré zmínit jak to ovlivňuje Git Flow

File: practices/git_flow.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+- gitflow
+- workflow
+- branch
+- master
+- develop
+- feature
+- release
+- hotfix
+---
+## What Is a Git Flow Workflow
```
Comment (prokopsimek) [inline practices/git_flow.md]: I'd prefer the name "Git Flow" only

File: practices/git_flow.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
+---
+Slug: git-flow-workflow
+Aspect: "processes"
+stages:
+- development
+Short_description: Git Flow Workflow is a branching system for Git. It helps the team to better control and add different project versions.
```
Comment (prokopsimek) [inline practices/git_flow.md]: I'd change `...is a branching system for Git.` to `...is opinionated a branching system for Git.` or `...is specific a branching system for Git.` to say that it's not the common workflow.

File: practices/git_flow.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+## Issues the Git Flow Workflow Solves
+- Unhappy clients
+- [Poor code quality](/issues/poor-code-quality)
+- Toxic team culture
+- Knowledge hoarding
+- Not my problem mentality
+- Meaningless work
+
+## How to Implement the Git Flow Workflow
+There are lots of different applications for Git Flow Workflow (such as [SmartGit](https://www.syntevo.com/smartgit/) or [GitKraken](https://www.gitkraken.com/). Pick the application suitable for your operation system to initialize the Git Flow. Customize your project setup by following these steps:
```
Comment (prokopsimek) [inline practices/git_flow.md]: The most used is SourceTree, please add it.

File: practices/git_flow.md (hunk 1 above)
Comment (prokopsimek) [reply practices/git_flow.md]: @pamichalik What do you think?

File: practices/git_flow.md (hunk 4 above)
Comment (prokopsimek) [reply practices/git_flow.md]: > `There are lots of different applications for Git Flow Workflow`

It's not about the application for Git Flow... You should say that all these Git applications supports Git Flow branching model and has a feature (specific settings) for that.

File: practices/git_flow.md (hunk 1 above)
Comment (pamichalik) [reply practices/git_flow.md]: regarding the text: "... team has too many features and is unable to finish them ..." I just wonder how this affects git flow?

File: practices/git_flow.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
+---
+Slug: git-flow
```
Comment (prokopsimek) [inline practices/git_flow.md]: must be small "S"

File: practices/git_flow.md [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+Aspect: "processes"
```
Comment (prokopsimek) [inline practices/git_flow.md]: must be small "A"

File: practices/git_flow.md [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+stages:
+- development
+Short_description: Git Flow is a specific branching system for Git. It helps the team to better control and add different project versions.
```
Comment (prokopsimek) [inline practices/git_flow.md]: must be small "S"

File: practices/git_flow.md [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+Tags:
```
Comment (prokopsimek) [inline practices/git_flow.md]: must be small "T"

File: practices/git_flow.md [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+- Git
+- gitflow
+- workflow
+- git-flow
+- flow
+- branch
+- branching
+Keywords:
```
Comment (prokopsimek) [inline practices/git_flow.md]: must be small "K"

File: practices/git_flow.md (hunk 1 above)
Comment (Kristynaslomkova) [reply practices/git_flow.md]: I meant that git flow affects the team when there are too many features. The team is not able to finish all of the features before the release.

Comment (prokopsimek) [reply practices/git_flow.md]: @Kristynaslomkova Could you add it to the article, please?

Comment (pamichalik) [reply practices/git_flow.md]: OK, now I understand

**Pull request merged**